### DISCUSSION TITLE
Provide dockerfile / dockerhub integration

### DISCUSSION THREAD

**Discussion by fabratu** [networkit/networkit#683] [Ideas]
To increase the user base (and decrease the entry bar) I think it might be interesting to integrate NetworKit into dockerhub. We could tie that automatically to different hooks like `latest` to a merge to `Dev` and version releases to `tags`. All we need for that would be (more or less) a `dockerfile` in the root-directory of the repository. What you think?

**Comment by avdgrinten** [comment]
Sounds good in general but I think that we need a use case first. How would a user use such a Docker image? Would he run his network analytics inside the container? Or in a container derived from the NetworKit one?

**Comment by fabratu** [comment]
A user can do both actually. Either he can work inside it directly or derive from it. In the case of deriving I would say the main use case is building a complete working environment (with additional packages and files), rather than adding custom code to NetworKit. 

A reason for using the vanilla-container I can see is zero-configuration time, since the image already can include dependencies like cython and examples from notebooks and graph-files. We could also add opt. packages like jupyter-notebook, but this of course would inflate the image and make it less minimal.
